File: 1702-maximum-binary-string-after-change/1702-maximum-binary-string-after-change.cpp
class Solution {
public:
    string maximumBinaryString(string binary) {
        int count = 0, j = -1;
        for(int i = binary.size() - 1; i >= 0; i--) {
            if(binary[i] == '0') {
                count++;
                j = i;
            }
            binary[i] = '1';
        }
        if(j != -1)
            binary[j + count - 1] = '0';
        return binary;
    }
};

File: 1702-maximum-binary-string-after-change/1702-maximum-binary-string-after-change.go
func maximumBinaryString(binary string) string {
    count := 0
    j := -1
    ans := make([]byte, len(binary));
    for i := len(binary) - 1; i >= 0; i-- {
        if binary[i] == '0' {
            count++
            j = i
        }
        ans[i] = '1'
    }
    if j != -1 {
        ans[j + count - 1] = '0';
    }
    return string(ans);
}